frontend/src/hooks/useDomainApi.ts

```diff
@@ -0,0 +1,55 @@
+import { Query, Domain } from 'types';
+import { useAuthContext } from 'context';
+import { useCallback } from 'react';
+
+interface ApiResponse {
+  result: Domain[];
+  count: number;
+}
+
+const PAGE_SIZE = 25;
+
+export const useDomainApi = (showAll: boolean) => {
```

Comment:
Can you explain your thinking for why it's good to factor useDomainApi into a separate hook? I noticed it's only used once in the code, at least now.

Are you thinking we can do something similar for the other API calls as well (vulnerabilities, etc.)?

Reply:
I think it makes sense to have all API access go through a layer like this. I only did the domain api because I started off looking at the Dashboard, but would eventually do others (vulnerabilities, etc.). What do you think?